### ISSUE TITLE
Extend atomic radii table

### ISSUE THREAD

**Issue by douweschulte** [BradyAJohnston/MolecularNodes#84]
While browsing through the code I saw that [your atomic radii table](https://github.com/BradyAJohnston/MolecularNodes/blob/main/src/import_structure.py#L182) is not fully populated. If you want you can reuse [the table](https://github.com/douweschulte/pdbtbx/blob/master/src/structs/elements.rs#L592) I generated for my own structural biochemistry projects. Be warned however here I put in four different definitions of atomic radii so it could be that this is a bit overkill for your application.

By the way thanks for the nice workshop on Tuesday, it really looks nice and was a nice way of getting to know this project. I and my colleague have been rendering all day today.

**Comment by BradyAJohnston** [comment]
Thanks for picking that up! I really need to go back and pretty much re-write everything internally. The code base is pretty messy and I made it solo without any history in python coding (lots of R though).

Will include the much more robust table in future versions I think. Would be very open to utilities your Rust library as well. I've had my eye on Rust for a while, and would love a chance to learn it. I don't know how it would go about interfacing with the python API of Blender, but if you have ideas of how it could be work I would be very interested!

Glad to hear you are having fun rendering lots of proteins as well, look forward to seeing some of the results!

**Comment by douweschulte** [comment]
I have done some googling and I did not find anyone who made a tutorial on how to combine Blender and Rust. But there is a lot of knowledge on how to combine Python with Rust.

* https://rustpython.github.io
* https://github.com/PyO3/pyo3
* https://github.com/dgrunwald/rust-cpython

Any of the last two can be used to generate Rust bindings for the Python Blender API, although I have never undertaken such a project before. If you want to look into this more I would be happy to help along.

**Comment by douweschulte** [comment]
I had a look through your code and it seems like you have a lot of back and forth between the blender api and your code, this makes it less logical to use a Rust binding to write everything in Rust, because every Rust <-> Python translation will incur some performance overhead. It is still possible but it could be that the overhead will be significant. But parsing the PDBs in Rust could be so much faster that it makes up for this fact.

Also more related to the original raised issue I found another place in your code where the radii are defined: https://github.com/BradyAJohnston/MolecularNodes/blob/main/src/find_bonds#L6.

And some pointers for Blender/Rust integration:
* https://github.com/italic-r/BlenderRust
* https://www.janwalter.org/jekyll/blender/rust/blendinfo/2019/05/28/blend_info.html

**Comment by BradyAJohnston** [comment]
Thanks for looking into this mate!

Certainly a pretty much complete re-write of the addon is on my list when I get home. The way it was previously structured (and my initial lack of python knowledge when I started making it) means there is a lot of redundant repeating of code, and also the are likely optimisations of the code to be made as well.

For the flow of data, the external python libraries that I currently use (`Atomium` for general PDB parsing, `MDAnalysis` for trajectory parsing) are only really used for the parsing of the files. I try to pass this data off to Blender as quickly as possible in the form of creating a 3D model inside of the Blender scene that can then actually store the data. 

For that case, if it would be quicker and more robust to parse PDB / mmCIF from the protein data bank itself, and local multi-PDB files from morphs created in PyMol using a Rust library (such as yours) then it would still be potentially worth it.

Once the data is inside of Blender, then everything is basically node based which are created inside of Blender anyway it's just extracting the data from .pdb files and trajectories into Blender which is currently the computationally difficult part. 

A pain point for users at the moment, is installation of `Atomium` itself. Currently I use `pip` from the internally bundled version pyhton kernal that is shipped with Blender. This proves problematic for installation in some cases. I have looked at bundling the required packages along with the addon but this blows the download size from ~ 1MB out to ~ 90MB, which I would like to avoid. 

If a Rust .pdb parser could solve this issue and be able to be included without the need for external installation calls, then this would also be a benefit switching the main parsing of the files. I haven't programmed with Rust yet, not have I looked in detail at the crate that you have made, so I am certainly open to suggestions and feedback on what you think a good approach would be. 

With a complete re-write of the addon on my todo list, for clarity of code and also to open it up to community contributions more which multiple people have expressed interest in doing, then choosing a new PDB parser (I'm currently looking at [biopandas](https://github.com/rasbt/biopandas) to replace Atomium) would be something that I would want to settle on before committing to doing so too heavily.
